### PR TITLE
Create CI

### PR DESCRIPTION
**Pull request by dfed**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: swift test -c release --enable-code-coverage -Xswiftc -enable-testing
+
+  readme-validation:
+    name: Check Markdown links
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Validate Markdown
+        uses: gaurav-nelson/github-action-markdown-link-check@v1
```
Comment (dfed) [inline .github/workflows/ci.yml]: far from a requirement but I've started running this on all of my repos. Catches broken README links in CI

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    name: Build and Test on Xcode 16
+    runs-on: macos-14
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Select Xcode Version
+        run: sudo xcode-select --switch /Applications/Xcode_16.app/Contents/Developer
+      - name: Build and Test Framework
+        run: |
+          xcrun swift test -c release --enable-code-coverage -Xswiftc -enable-testing
```
Comment (dfed) [inline .github/workflows/ci.yml]: I don't have permissions to set up Codecov in this repo so I didn't set that up. We’ll Get There™

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  linux:
+    name: Build and Test on Linux
+    runs-on: ubuntu-latest
+    container: swift:6.0
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Build and Test Framework
+        run: swift test -c release --enable-code-coverage -Xswiftc -enable-testing
```
Comment (dfed) [inline .github/workflows/ci.yml:30]: this is becoming a habit of mine in open source libs: test in release!

makes things a bit slower in CI but I have caught a few esoteric issues this way.

Comment (dfed) [reply .github/workflows/ci.yml:30]: though testing in release on macOS yields an error executing the CLI, which... I don't get why. works on linux at least.

Comment (fdiaz) [reply .github/workflows/ci.yml:30]: Oh interesting! I've never thought about testing in release. Let's try it out!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fdiaz) [reply .github/workflows/ci.yml]: Sounds like we should be using [LinkInspector](https://github.com/UmbrellaDocs/action-linkspector) instead according to the README of [the repo](https://github.com/gaurav-nelson/github-action-markdown-link-check).

Sounds like this one is being deprecated in favor of `LinkIspector`.

Comment (dfed) [reply .github/workflows/ci.yml]: oh man. I gotta update this everywhere 😅

File: .github/workflows/ci.yml (hunk 2 above)
Comment (fdiaz) [reply .github/workflows/ci.yml]: https://github.com/fdiaz/SafeDI-Formatter/issues/6

**Pull request merged**